what4/src/What4/Protocol/SMTWriter.hs

```diff
@@ -786,6 +786,10 @@ class (SupportTermOps (Term h)) => SMTWriter h where
   -- | Reset the solver state, forgetting all pushed frames and assertions
   resetCommand  :: f h -> Command h
 
+  -- | Set the timeout for the next goal. Returns @Nothing@ if setting
+  -- per-goal timeouts is not supported.
+  setGoalTimeoutCommand  :: f h -> Integer -> Maybe (Command h)
+
```

Comment:
My only request is that we either use a more structured type for the timeout, or at the very least document the units here (in case some solvers use milliseconds vs seconds or something).

Reply:
Good point. Maybe it should be a `DiffTime`?

Reply:
That is kind of appealing, but maybe it isn't worth an extra dependency... on the other hand, it seems highly likely that what4 already depends on something that indirectly depends on `time`

Reply:
From a quick look, `crucible`, `saw-core`, `cryptol`, `llvm-pretty-bc-parser`, and `crux` all depend on `time`.

Reply:
Oh, and so does `directory`, which `what4` directly depends on. And it's a base package included with GHC.